### PR TITLE
Avro schema support

### PR DESCRIPTION
**Pull request by willyborankin**
1. Add support for Avro schema
2. Bump up libs
3. Add integration test for Avro

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -74,17 +78,26 @@ configurations {
 
 dependencies {
     compileOnly "org.apache.kafka:connect-api:$kafkaVersion"
-    implementation 'org.slf4j:slf4j-api:1.7.28'
+    compileOnly "org.apache.kafka:connect-json:$kafkaVersion"
+    compileOnly "org.apache.avro:avro:1.8.1"
```
Comment (ivanyu) [inline build.gradle:82]: Seems not needed

File: build.gradle [hunk 2]
```diff
@@ -74,17 +78,26 @@ configurations {
...
-    testImplementation 'org.hamcrest:hamcrest:2.1'
+    testImplementation "org.junit.jupiter:junit-jupiter:5.5.2"
+    testImplementation "org.hamcrest:hamcrest:2.1"
     testImplementation "org.apache.kafka:connect-api:$kafkaVersion"
-    testRuntime 'org.apache.logging.log4j:log4j-slf4j-impl:2.12.1'
+    testRuntime "org.apache.logging.log4j:log4j-slf4j-impl:2.12.1"
+    testRuntime "com.fasterxml.jackson.core:jackson-databind:2.9.10"
+    testRuntime "org.apache.kafka:connect-json:$kafkaVersion"
+
+    testImplementation "org.apache.avro:avro:1.8.1"
```
Comment (ivanyu) [inline build.gradle:93]: Seems not needed

File: src/integration-test/java/io/aiven/kafka/connect/http/IntegrationTest.java [hunk 3]
```diff
@@ -285,7 +285,7 @@ final void testBatching() throws ExecutionException, InterruptedException {
                 sendFutures.add(sendMessageAsync(TEST_TOPIC, partition, key, value));
 
                 batchRecordCnt += 1;
-                currentBody.append(value).append("\n");
+                currentBody.append(value).append(BatchRecordSender.BATCH_RECORD_SEPARATOR);
```
Comment (ivanyu) [inline src/integration-test/java/io/aiven/kafka/connect/http/IntegrationTest.java:288]: There are a couple of reasons to use a literal in places like this: 1) not expose privates; 2) actually see what we're testing and not relying on whatever (potentially erroneous) value we have in the constant.

File: src/main/java/io/aiven/kafka/connect/http/converter/RecordValueConverter.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    private final Map<Class<?>, Converter> converters = new HashMap<>();
+
+    interface Converter {
+        String convert(final SinkRecord record);
+    }
+
+    public String convert(final SinkRecord record) {
+        return converters
+                .computeIfAbsent(record.value().getClass(), this::createConverter)
```
Comment (ivanyu) [inline src/main/java/io/aiven/kafka/connect/http/converter/RecordValueConverter.java:36]: Looks like something that should be just eagerly init in the constructor.

**Review by ivanyu — CHANGES_REQUESTED** [review]
Good thing! Just a couple of minor things

File: src/integration-test/java/io/aiven/kafka/connect/http/IntegrationTest.java (hunk 3 above)
Comment (willyborankin) [reply src/integration-test/java/io/aiven/kafka/connect/http/IntegrationTest.java:288]: will bring it back

File: src/main/java/io/aiven/kafka/connect/http/converter/RecordValueConverter.java (hunk 4 above)
Comment (willyborankin) [reply src/main/java/io/aiven/kafka/connect/http/converter/RecordValueConverter.java:36]: Good point!

**Review by ivanyu — COMMENTED** [review]
LGTM. Planning to do some squash to commits?

**Comment by willyborankin** [comment]
@ivanyu squashed

**Review by ivanyu — APPROVED** [review]

**Pull request merged**